src/Interpreters/QueryMetricLog.cpp

```diff
@@ -0,0 +1,180 @@
+#include <base/getFQDNOrHostName.h>
+#include <Common/CurrentThread.h>
+#include <Common/DateLUT.h>
+#include <Common/DateLUTImpl.h>
+#include <DataTypes/DataTypeDate.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypeDateTime64.h>
+#include <DataTypes/DataTypeLowCardinality.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Interpreters/Context.h>
+#include <Interpreters/QueryMetricLog.h>
+#include <Interpreters/PeriodicLog.h>
+#include <Interpreters/ProcessList.h>
+#include <Parsers/ExpressionElementParsers.h>
+#include <Parsers/parseQuery.h>
+
+#include <chrono>
+#include <mutex>
+
+namespace CurrentMetrics
+{
+    extern const Metric MemoryTracking;
+    extern const Metric MergesMutationsMemoryTracking;
+}
+
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+}
+
+namespace DB
+{
+
+const auto memory_metrics = std::array{CurrentMetrics::MemoryTracking, CurrentMetrics::MergesMutationsMemoryTracking};
+
+ColumnsDescription QueryMetricLogElement::getColumnsDescription()
+{
+    ColumnsDescription result;
+    ParserCodec codec_parser;
+
+    result.add({"hostname",
+                std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()),
+                parseQuery(codec_parser, "(ZSTD(1))", 0, DBMS_DEFAULT_MAX_PARSER_DEPTH, DBMS_DEFAULT_MAX_PARSER_BACKTRACKS),
+                "Hostname of the server executing the query."});
+    result.add({"event_date",
+                std::make_shared<DataTypeDate>(),
+                parseQuery(codec_parser, "(Delta(2), ZSTD(1))", 0, DBMS_DEFAULT_MAX_PARSER_DEPTH, DBMS_DEFAULT_MAX_PARSER_BACKTRACKS),
+                "Event date."});
+    result.add({"event_time",
+                std::make_shared<DataTypeDateTime>(),
+                parseQuery(codec_parser, "(Delta(4), ZSTD(1))", 0, DBMS_DEFAULT_MAX_PARSER_DEPTH, DBMS_DEFAULT_MAX_PARSER_BACKTRACKS),
+                "Event time."});
+    result.add({"event_time_microseconds",
+                std::make_shared<DataTypeDateTime64>(6),
```

Comment:
Why doesn't this column has a codec and the others do? Since event_time_microseconds will increase in a similar fashion as event_time it'd make sense to have a similar codec. OTOH, I'm not sure if codecs are useful at all here, and it might be that ZSTD by itself is more than enough

Reply:
The answer is that no other `event_time_microseconds` for any other log had any codec previously. The rest of the rows here have codecs since I created [ErrorLog](https://github.com/ClickHouse/ClickHouse/issues/64501), but prior to that one none had codecs.

I'll add the `Delta(4), ZSTD(1)` codec for consistency, in any case :+1: